### PR TITLE
Validate the 'shaper', 'acl' and 'access' sections

### PR DESCRIPTION
**Pull request by chrzaszcz**
Parse the 'acl' section in a more structured way,
matching exact rules from the 'acl' module.

### CODE REVIEW

File: src/config/mongoose_config_parser_toml.erl [hunk 1]
```diff
@@ -1365,21 +1365,44 @@ process_acl([ACLName, _|Path], Content) ->
...
+    {AclName, AclKeys} = find_acl(M, lists:sort(maps:keys(M)), acl_keys()),
+    list_to_tuple([AclName | lists:map(fun(K) -> maps:get(K, M) end, AclKeys)]).
+
+find_acl(M, SortedMapKeys, [{AclName, AclKeys}|Rest]) ->
+    case lists:sort(AclKeys) of
+        SortedMapKeys -> {AclName, AclKeys};
+        _ -> find_acl(M, SortedMapKeys, Rest)
+    end.
+
+acl_keys() ->
```
Comment (gustawlippa) [inline src/config/mongoose_config_parser_toml.erl:1380]: In `acl` module there is also `resource`, isn't it missing?

Comment (chrzaszcz) [reply src/config/mongoose_config_parser_toml.erl:1380]: Good catch!

File: src/config/mongoose_config_parser_toml.erl [hunk 2]
```diff
@@ -1365,21 +1365,44 @@ process_acl([ACLName, _|Path], Content) ->
...
-            list_to_tuple([b2a(Key) | Values]);
-        [{Key, Value}] when is_binary(Value) ->
-            {b2a(Key), Value}
-    end;
-acl_data(Value) when is_binary(Value) -> b2a(Value).
+acl_data(<<"all">>) -> all;
+acl_data(<<"none">>) -> none;
+acl_data(M) ->
+    {AclName, AclKeys} = find_acl(M, lists:sort(maps:keys(M)), acl_keys()),
+    list_to_tuple([AclName | lists:map(fun(K) -> maps:get(K, M) end, AclKeys)]).
```
Comment (gustawlippa) [inline src/config/mongoose_config_parser_toml.erl:1372]: If I understand this line correctly, I would find something like this more readable, but it's a matter of taste.
```suggestion
    list_to_tuple([AclName | maps:to_list(maps:with(AclKeys, M))).
```

Comment (chrzaszcz) [reply src/config/mongoose_config_parser_toml.erl:1372]: Not quite the same, your version changes the ordering of the keys.

**Pull request merged**